src/tests/test_utils/common_test_utils/src/ov_plugin_cache.cpp

```diff
@@ -34,14 +34,15 @@ PluginCache& PluginCache::get() {
 }
 
 std::shared_ptr<ov::Core> PluginCache::core(const std::string& deviceToCheck) {
-    std::lock_guard<std::mutex> lock(g_mtx);
     if (std::getenv("DISABLE_PLUGIN_CACHE") != nullptr) {
 #ifndef NDEBUG
         std::cout << "'DISABLE_PLUGIN_CACHE' environment variable is set. New Core object will be created!"
                   << std::endl;
 #endif
         return std::make_shared<ov::Core>();
     }
+
+    std::lock_guard<std::mutex> lock(g_mtx);
```

Comment:
`register_plugin()` has it's own mutex. So should we use this lock? If only `register_plugin()` is called, I suppose lock not required, and `ov_core` shouldn't be changed simultaneously